### PR TITLE
Redstone Recipes

### PR DESCRIPTION
**Pull request by Rakkis157**
Working on adding recipes to craft redstone components, excluding railway stuff since I will be doing those in another batch.

### CODE REVIEW

**Comment by Azercoco** [comment]
I will juste generate all the wood recipe with the Datagen, it will be far easier

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/packer/iron_door.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "type": "modern_industrialization:packer",
+  "eu": 2,
+  "duration": 100,
+  "item_inputs": [
+    {
+      "item": "modern_industrialization:iron_plate",
+      "amount": 3
+    },
+    {
+      "item": "modern_industrialization:iron_bolt",
```
Comment (Azercoco) [inline src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/packer/iron_door.json:11]: Would remove the bolt as they are not mean to be useful after the steam age ... but it risk to cause conflict

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/piston.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {
+      "tag": "minecraft:planks",
+      "amount": 3
+    },
+    {
+      "item": "minecraft:redstone",
+      "amount": 1
+    },
+    {
+      "item": "modern_industrialization:steel_rod",
```
Comment (Azercoco) [inline src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/piston.json:19]: Iron rod maybe ?

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "type": "modern_industrialization:assembler",
+  "eu": 8,
+  "duration": 200,
+  "item_inputs": [
+    {
+      "item": "minecraft:cobblestone",
+      "amount": 7
+    },
+    {
+      "item": "minecraft:string",
```
Comment (Azercoco) [inline src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json:11]: Conflicts with the dropper

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{
+  "type": "modern_industrialization:assembler",
+  "eu": 8,
+  "duration": 200,
+  "item_inputs": {
+      "tag": "minecraft:planks",
```
Comment (Azercoco) [inline src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json:6]: Will conflict with a lot of thing

**Review by Azercoco — CHANGES_REQUESTED** [review]

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json (hunk 3 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json:11]: Hmm... Gonna make the recipes into
Dropper:
6 Cobble
1 Redstone
1 Bronze Ring

Dispenser
6 Cobble
1 Redstone
1 String
1 Iron Ring
Basically give them nozzles.

Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json:11]: What rings to use is not final. Considering using Invar for one tho.

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json (hunk 4 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json:6]: Yeah, I'll try to cook up something for it.

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/piston.json (hunk 2 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/piston.json:19]: Sure.

**Comment by Rakkis157** [comment]
> I will juste generate all the wood recipe with the Datagen, it will be far easier

That would be great.

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/packer/iron_door.json (hunk 1 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/packer/iron_door.json:11]: I figured that for the more niched items, using early game stuff wouldn't be too big of an issue. Players that aim to automate stuff like iron doors, whether for building or out of completionism, wouldn't mind throwing on an extra machine or two, while the players who are in it purely for progression aren't too affected.

Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/packer/iron_door.json:11]: That said, made the iron bolt for the trapdoor into an iron ring, to reduce recipe conflicts.

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json (hunk 3 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/dispenser.json:15]: Done.

File: src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json (hunk 4 above)
Comment (Rakkis157) [reply src/main/resources/data/modern_industrialization/recipes/vanilla_recipes/assembler/chest.json:7]: Went with adding iron parts into the chest.
2 bolts to represent hinges and an iron ring for the clasp.

**Review by Technici4n — CHANGES_REQUESTED** [review]
The wood variants need to be done with datagen.

**Comment by Azercoco** [comment]
I will close this issue as I think it's not very useful in MI and aligned with the mod focus and has a too high probability of causing recipe conflict.

**Pull request closed without merging**